api/filter_dimensions.go

```diff
@@ -89,10 +89,23 @@ func (api *API) addFilterDimension(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	dimensions, err := api.validateAndHydrateDimensions(v, []model.Dimension{req.Dimension}, filter.PopulationType)
-	if err != nil {
-		api.respond.Error(ctx, w, statusCode(err), errors.Wrap(err, "failed to validate dimensions"))
-		return
+	var dimensions []model.Dimension
+
+	if v.IsBasedOn.Type == cantabularFlexibleTable {
+		dimensions, err = api.validateAndHydrateDimensions(v, []model.Dimension{req.Dimension}, filter.PopulationType)
+		if err != nil {
+			api.respond.Error(ctx, w, statusCode(err), errors.Wrap(err, "failed to validate dimensions"))
+			return
+		}
+	} else if v.IsBasedOn.Type == cantabularMultivariateTable {
```

Comment:
Won't this force paths not from POST /dimensions down this way when we don't want them to be? The existing checks for multivariates from other endpoints were correct no? This isn't the best way to be attempting to do this.